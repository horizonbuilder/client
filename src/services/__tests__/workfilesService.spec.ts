import WorkfilesService from '../workfiles';

const workfilesData = [
  {
    name: 'workfile0',
    status: 'in-progress',
    client_name: 'client0',
    created_at: '2017-09-22T11:12:55.325Z'
  },
  {
    name: 'workfile1',
    status: 'prospective',
    client_name: 'client1',
    created_at: '2017-09-22T11:12:55.330Z'
  }
];

describe('WorkfilesService', () => {
  beforeEach(() => {
    global.fetch = null;
  });

  describe('#getWorkfiles', () => {
    test('returns an array of workfiles', async () => {
      global.fetch = jest.fn(async () => ({
        json: async () => workfilesData
      }));

      const workfiles = await WorkfilesService.getWorkfiles();

      expect(workfiles).toMatchSnapshot();
    });
  });
});
