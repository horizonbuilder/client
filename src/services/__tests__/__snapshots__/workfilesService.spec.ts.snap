// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkfilesService #getWorkfiles returns an array of workfiles 1`] = `
Array [
  Object {
    "client_name": "client0",
    "created_at": "2017-09-22T11:12:55.325Z",
    "name": "workfile0",
    "status": "in-progress",
  },
  Object {
    "client_name": "client1",
    "created_at": "2017-09-22T11:12:55.330Z",
    "name": "workfile1",
    "status": "prospective",
  },
]
`;
